### PR TITLE
Fix restart loop by making sure gamedata has changed

### PR DESCRIPTION
**Pull request by roy7**
Doing same thing as https://github.com/online-go/gtp2ogs/pull/171 but leaving out the other changes @windo was still testing.

### CODE REVIEW

**Comment by wonderingabout** [comment]
great ! :100:

**Comment by roy7** [comment]
@Dorus From Discord discussion on recent gamedata packet problems, it came up that since

`this.state = gamedata;`

is before 

`const gamedataChanged = (JSON.stringify(this.state) !== JSON.stringify(gamedata));`

how does it ever fail this test?

File: game.js [hunk 1]
```diff
@@ -82,8 +82,16 @@ class Game {
             // restart the bot by killing it here if another gamedata comes in. There normally should only be one
             // before we process any moves, and makeMove() is where a new Bot is created.
             //
-            if (this.bot) {
-                this.log("Killing bot because of gamedata packet after bot was started");
+            let gamedataChanged = (JSON.stringify(this.state) !== JSON.stringify(gamedata));
```
Comment (roy7) [inline game.js:85]: Does this need to be before line 47?

Comment (wonderingabout) [inline game.js:85]: could you use `const` instead of `let`

Comment (wonderingabout) [reply game.js:85]: edit: nevermind it seems this is not a current pr

**Pull request merged**